frontend/src/app/shared/components/forms/checkbox-field/checkbox-field.component.html

```diff
@@ -0,0 +1,45 @@
+<ng-container *ngIf="!hidden">
+  <label class="op-form-field--label-wrap op-checkbox-field--label-wrap">
```

Comment:
Out of curiosity: I totally see the benefit of having an angular component, but why did you create your own HTML-tree (and classes) instead of using at least some of the class we already have? None of the the required designs is really new. We already have checkboxes with labels on the right side and an explanatory text below. You can have a look here: http://localhost:3000/assets/styleguide.html#forms

Reply:
`op-form-field`, `op-form`, and `op-option-list` were created for the add user modal, where for the first time we had a set of related style changes:

* Label above the field instead of next to it
* Validation errors are shown inline

I didn't (and don't) think it would've been a great idea to e.g. have a modifier on the old `form--field` styles, because the existing styles are not in standard BEM style and really hard to refactor. (`frontend/src/global_styles/content/_forms.sass` is 900 lines of sass) They work well in the current usecases, but I figured instead of trying to put changes in there potentially breaking some far away forms, I'd create components specific to the new styles.

These were then adapted to work with the formly dynamic forms that were used on the project settings. Here `op-form` was also finalized a tad more.

Here I created `op-checkbox-field` because I didn't want to blow up `op-form-field` with more checkbox-specific changes. `op-form-field` expects the input to be below the label and some other specialties, however in this design the checkboxes came before the text. As you can see, `op-checkbox-field` still expects to be based on an `op-form-field`, but overwrites and rearranges the children https://github.com/opf/openproject/pull/9636/files/f3aa8efa4fa7dc138fad2c7fb7f4e86362ea7b8f#diff-9a31b6533922f486f0af10580880dd636098f43d91c4620047a12cec442d0eb3R3

This is a lot more than what used to be there for checkboxes; https://github.com/opf/openproject/pull/9636/files/f3aa8efa4fa7dc138fad2c7fb7f4e86362ea7b8f#diff-d79513a013f1fa2f0bcc4d8f12b3faff7180f047665fdd4d8d15527a2842c521L54 which was code that expected the same HTML structure as for `op-form-field`. However ,the two HTML structures now slightly differ.

tl;dr Instead of bloating`form--field` with `op-form-field` and  `op-form-field` with more versions I opted for this. I'd be okay with saying no, `op-checkbox-field` (the BEM block) should be a modifier of `op-form-field`, but I wouldn't try to adapt the old styles to the new layouts or add more styles modifiers there.

Reply:
That makes totally sense for me. Thanks for the detailed explanation 👍